corehq/apps/dhis2/models.py

```diff
@@ -111,156 +115,17 @@ def get_dataset(self):
-                'Request details: %s\n'
-                'Error: %s',
-                {'method': 'delete', 'url': self.server_url + path, 'headers': self.headers},
-                err)
-            raise JsonApiError(str(err))
-        return JsonApiRequest.json_or_error(response)
-
-    def post(self, path, data, **kwargs):
-        # Make a copy of self.headers because setting content type on requests that don't send content is bad
-        headers = self.headers.copy()
-        headers['Content-type'] = 'application/json'
-        json_data = json.dumps(data, default=json_serializer)
-        logging.debug(
-            'DHIS2: POST %s: \n'
-            '    Headers: %s\n'
-            '    Data: %s\n'
-            '    kwargs: %s',
-            self.server_url + path, self.headers, json_data, kwargs
-        )
-        try:
-            response = requests.post(self.server_url + path, json_data, headers=headers, auth=self.auth, **kwargs)
-        except requests.RequestException as err:
-            logging.exception(
-                'JSON API raised HTTP or socket error.\n'
-                'Request details: %s\n'
-                'Error: %s',
-                {'method': 'post', 'url': self.server_url + path, 'data': json_data, 'headers': headers},
-                err
-            )
-            raise JsonApiError(str(err))
-        return JsonApiRequest.json_or_error(response)
-
-    def put(self, path, data, **kwargs):
-        headers = self.headers.copy()
-        headers['Content-type'] = 'application/json'
-        json_data = json.dumps(data, default=json_serializer)
-        logging.debug(
-            'DHIS2: PUT %s: \n'
-            '    Headers: %s\n'
-            '    Data: %s\n'
-            '    kwargs: %s',
-            self.server_url + path, self.headers, json_data, kwargs
-        )
-        try:
-            response = requests.put(self.server_url + path, json_data, headers=headers, auth=self.auth, **kwargs)
-        except requests.RequestException as err:
-            logging.exception(
-                'JSON API raised HTTP or socket error.\n'
-                'Request details: %s\n'
-                'Error: %s',
-                {'method': 'put', 'url': self.server_url + path, 'data': json_data, 'headers': headers},
-                err
-            )
-            raise JsonApiError(str(err))
-        return JsonApiRequest.json_or_error(response)
+    domain = models.CharField(max_length=126, db_index=True)  # 126 seems to be a popular length
+    timestamp = models.DateTimeField()
+    request_method = models.CharField(max_length=12)
+    request_url = models.CharField(max_length=255)
+    request_headers = jsonfield.JSONField(blank=True)
+    request_params = jsonfield.JSONField(blank=True)
+    request_body = jsonfield.JSONField(blank=True)
+    request_error = models.TextField(blank=True)
+    response_status = models.IntegerField()
+    response_body = models.TextField(blank=True)
```

Comment:
Does Dimagi have a position in the NULL vs empty string debate? i.e. Which (one or more) of the following is true:
- [ ] NULL == "I don't know" 
- [ ] NULL == "I do know and it's not applicable" 
- [ ] NULL == "I do know and it has no value"

I'm happy to adopt whatever convention we've chosen. Your comment makes me think that our convention is either "True, True, True" or "True, True, False". For now I'll assume "True, True, False" and change this accordingly.

If we don't have a convention, I'll take the initial position of "True, I'm open to persuasion, False." 

If the convention is "True, False, False" then there would be no change because all the values are known. If it's "True, True, (True|False)" then all those properties should be nullable (including `response_status` if the request errors before it is sent).